services/web-app/utils/use-query-state.js

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright IBM Corp. 2021, 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { useRouter } from 'next/router'
+import { useCallback, useEffect, useMemo } from 'react'
+
+export const queryTypes = {
+  string: {
+    parse: (v) => v,
+    serialize: (v) => `${v}`
+  },
+  integer: {
+    parse: (v) => parseInt(v),
+    serialize: (v) => Math.round(v).toFixed()
+  },
+  float: {
+    parse: (v) => parseFloat(v),
+    serialize: (v) => v.toString()
+  },
+  boolean: {
+    parse: (v) => v === 'true',
+    serialize: (v) => (v ? 'true' : 'false')
+  },
+  timestamp: {
+    parse: (v) => new Date(parseInt(v)),
+    serialize: (v) => v.valueOf().toString()
+  },
+  isoDateTime: {
+    parse: (v) => new Date(v),
+    serialize: (v) => v.toISOString()
+  }
+}
+
+export const useQueryState = (
+  key,
+  {
+    defaultValue = '',
+    parse = queryTypes.string.parse,
+    serialize = queryTypes.string.serialize,
+    useStorage = false
```

Comment:
minor: naming convention on this boolean makes it look like a react hook. Maybe something like `shouldUseStorage` instead?

Reply:
I'll update to `saveToStorage`.